jcli/src/jcli_lib/vote/tally/merge_results.rs

```diff
@@ -0,0 +1,344 @@
+use crate::jcli_lib::utils::io;
+use crate::jcli_lib::utils::OutputFormat;
+use jormungandr_lib::crypto::hash::Hash;
+use jormungandr_lib::interfaces::VotePlanId;
+use jormungandr_lib::interfaces::{PrivateTallyState, Tally, VotePlanStatus, VoteProposalStatus};
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::ops::Range;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("voteplan should be already decrypted before merging")]
+    VotePlanEncrypted,
+    #[error("voteplans have different privacy type")]
+    PrivacyMismatch,
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct MergeVotePlan {
+    /// The path to json-encoded list of voteplans to merge. If this parameter is not specified, it
+    /// will be read from the standard input. Voteplans must be already decrypted before merging.
+    /// Two voteplans in the list will be merged if they have ALL the same proposals according to
+    /// the proposal (external) id.
+    #[structopt(long)]
+    vote_plans: Option<PathBuf>,
+    #[structopt(flatten)]
+    output_format: OutputFormat,
+}
+
+impl MergeVotePlan {
+    pub fn exec(&self) -> Result<(), super::Error> {
+        let voteplans: Vec<VotePlanStatus> =
+            serde_json::from_reader(io::open_file_read(&self.vote_plans)?)?;
+
+        let results = merge_voteplans(voteplans)?;
+
+        let output = self
+            .output_format
+            .format_json(serde_json::to_value(results)?)?;
+        println!("{}", output);
+
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+pub struct MergedVotePlan {
+    pub ids: BTreeSet<VotePlanId>,
+    pub proposals: Vec<MergedVoteProposalStatus>,
+}
+
+/// like VoteProposalStatus but without the index field, since the proposals can be in different
+/// indexes with the current implementation.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct MergedVoteProposalStatus {
+    pub proposal_id: Hash,
+    pub options: Range<u8>,
+    pub tally: Tally,
+    pub votes_cast: usize,
+}
+
+fn merge_voteplans(mut voteplans: Vec<VotePlanStatus>) -> Result<Vec<MergedVotePlan>, Error> {
```

Comment:
this is more complicated than I would have expected, especially dealing with raw indexes. 
Can we partition the voteplans based on proposals right at the start?
Smth like:
```
struct WrappedVotePlan(VotePlansStatus);

impl PartialEq for WrappedVotePlanStatus {
    fn eq(&self, other: &Self) -> bool {
        let p1 = self.proposals.sort() // it's in-place in practice, but whatever...
        let p2 = other.proposals.sort()
        p1 == p2
    }
}

impl Hash for WrapperVotePlan {
    fn hash(&self, hasher: &mut Hasher) {
        hasher.input(self.proposals.sort())
    }
}

fn merge_voteplans(mut voteplans: Vec<VotePlanStatus>) -> ... {
    // HashMap<WrappedVotePlan, Vec<WrapperVotePlan>>
    let vps = voteplans.into_iter().map(WrapperVotePlan).fold(|mut acc, vp| acc.entry(vp.clone()).or_default().push(vp) );
    for (_, vps) in vps.into_iter() {
        do_merge(vps)
    }
}
```